src/scripts/build_dataset.py

```diff
@@ -29,6 +29,18 @@ def main(config: DictConfig) -> None:
     prompt_and = config.language.prompt_and
     clues_dict = config.language.clues_dict
 
+    n_red_herring_clues = config.n_red_herring_clues
+    red_herring_clues_dict = config.language.red_herring_clues_dict
+    red_herring_attributes = config.language.red_herring_attributes
+    red_herring_facts = config.language.red_herring_facts
+
+    red_herring_info = (
+        n_red_herring_clues,
+        red_herring_clues_dict,
+        red_herring_attributes,
+        red_herring_facts,
+    )
```

Comment:
To avoid silent errors, I would convert this into a dictionary instead, or simply have them as 4 separate arguments in `build_dataset`.